src/main/java/com/ea/restaurant/util/GrpcUtil.java

```diff
@@ -0,0 +1,35 @@
+package com.ea.restaurant.util;
+
+import com.ea.restaurant.constants.Authentication;
+import com.ea.restaurant.exceptions.Unauthenticated;
+import com.ea.restaurant.grpc.interceptors.GrpcServerInterceptor;
+import io.grpc.Metadata;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class GrpcUtil {
+
+  public static List<String> getCredentialsFromBasicAuthToken(Metadata metadata) {
+    var authorizationValue =
+        metadata.get(
+            Metadata.Key.of(Authentication.METADATA_KEY_NAME, Metadata.ASCII_STRING_MARSHALLER));
+
+    var authorizationToken =
+        Optional.ofNullable(authorizationValue)
+            .map(auth -> authorizationValue.substring("Basic".length()))
+            .orElseThrow(Unauthenticated::new);
+    var authorizationList = Stream.of(authorizationToken.trim()).toList();
+
+    var credentialsDecoded = Base64.getDecoder().decode((authorizationList.get(0)));
+    var credentials = new String(credentialsDecoded, StandardCharsets.UTF_8);
+    return Stream.of(credentials.split(":")).collect(Collectors.toList());
```

Comment:
Return a java record instead, you don't need to return a list, this will make life harder to the developer

Reply:
Done.